sympy/functions/special/bessel.py

```diff
@@ -586,6 +586,19 @@ def _eval_nseries(self, x, n, logx, cdir=0):
 
         return super(besseli, self)._eval_nseries(x, n, logx, cdir)
 
+    def _eval_aseries(self, n, args0, x, logx):
+        from sympy.functions.combinatorial.factorials import RisingFactorial
+        from sympy.series.order import Order
+        nu, z = self.args
+        point = args0[1]
+
+        if point is S.Infinity:
+            s = [(RisingFactorial(S.Half-nu,k)*RisingFactorial(
+                S.Half+nu,k))/((2)**(k)*z**((2*k+1)/2)*factorial(k)) for k in range(n)] + [Order(1/z**((2*n+1)/2), x)]
+            return (exp(z)/sqrt((2*pi))) * Add(*s)
+
+        return super()._eval_aseries(n, args0, x, logx)
+
```

Comment:
I think I had worked on a solution for this in my gsoc proposal a couple years back (though I wasn't able to implement it)
Would you like to have a look ? 
I think that might give you some direction for this PR. You can find my proposal [here](https://drive.google.com/file/d/1P8PFVoSyTcskkRVYmbHd-85o96dOiEnZ/view?pli=1)

Reply:
If we fix the above decrepency with something like this 
```diff
--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -3504,7 +3504,10 @@ def as_coeff_exponent(self, x) -> tuple[Expr, Expr]:
         c, p = s.as_coeff_mul(x)
         if len(p) == 1:
             b, e = p[0].as_base_exp()
-            if b == x:
+            # if b == x:
+            if b in [x, 1/x]:
+                if b == 1/x:
+                    e = -e
                 return c, e
         return s, S.Zero

```
we have the test passing alright.